src/models/json.ts

```diff
@@ -0,0 +1,37 @@
+import { BigNumber } from '@ethersproject/bignumber'
+
+type Primitive = string | number | boolean | symbol | null | undefined
+
+export type Json<T> = {
```

Comment:
new helper type that is used in a lot of places now. it recursively converts an object to typed key-values, where primitive value types are retained and more complex types are reduced to a JSON equivalent (BigNumber -> string, array -> typed array)

Reply:
as an example:

```
type Stuff = {
  a: string
  b: number
  c: string | null
  d: string[]
  e: BigNumber
  f: BigNumber[]
  g: {
    a: string
    b: number
    c: BigNumber
  }
}

type Json<Stuff> => {
  a: string
  b: number
  c: string | null
  d: string[]
  e: string
  f: string[]
  g: {
    a: string
    b: number
    c: string
  }
}
```